### ISSUE TITLE
1.11.0 release candidate, old versions in npm-shrinkwrap.json

### ISSUE THREAD

**Issue by john7doe** [appium/appium#11967]
## The problem

Looking at https://github.com/appium/appium/blob/1.11/npm-shrinkwrap.json it looks like it locks the version of the appium-* packages to some really old ones.

e.g. appium-adb is locked to version 6.19.0, when looking at the appium-adb repo I would have expected a ^7.0 version. The 1.10 release used 6.28.0 of the package.

Another example is appium-espresso-driver that is locked to 1.3 and the latest version released is 1.6.

**Comment by imurchie** [comment]
Yes, something odd seems to have gone on here. On the bright side, the package is also not set up to be distributed with the `npm-shrinkwrap.json` file included, so you'll get the most recent dependencies anyway. :|

@dpgraham Can you have a look at this?

**Comment by john7doe** [comment]
I was assuming that you were shipping with a `npm-shrinkwrap.json` file included?!?! 

If I understand npm correctly then, with the quite lax versions in `package.json`, there is nothing thats will  ensures that yesterdays `npm install appium@1.10` is the same as todays `npm install appium@1.10` ?

**Comment by dpgraham** [comment]
I can look at that. I think I probably didn't clear the node modules and reinstall before ahronkwrapping.

**Comment by john7doe** [comment]
1.9.0 did, it does not look like 1.10 or the latest rc did:
```
~/tmp $ curl -s https://registry.npmjs.org/appium/-/appium-1.11.0-rc.0.tgz | tar tv|grep npm-shrinkwrap.json
~/tmp $ curl -s https://registry.npmjs.org/appium/-/appium-1.10.0.tgz | tar tv |grep npm-shrinkwrap.json
~/tmp $ curl -s https://registry.npmjs.org/appium/-/appium-1.9.0.tgz | tar tv |grep npm-shrinkwrap.json
-rw-r--r--  0 0      0      453599 Oct 26  1985 package/npm-shrinkwrap.json
```

**Comment by dpgraham** [comment]
@john7doe you're right, I never noticed.

@imurchie looks like `npm-shrinkwrap.json` is being ignored (https://github.com/appium/appium/pull/11461).

**Comment by dpgraham** [comment]
I find it perpelixing that it's even _possible_ to ignore `npm-shrinkwrap.json`. It should be published and included by default.

**Comment by dpgraham** [comment]
Also, I published a new rc (with shrinkwrap) and it has not outdated dependencies.

**Comment by john7doe** [comment]
Thank you for moving on this so fast. I've had a look at v1.11.0-rc.1 and it looks much better :-) One (minor) thing still looks odd in `npm-shrinkwrap.json`:


```json
{
  "name": "appium",
  "version": "1.11.0-rc.0",
   ...
```

I would expect the version to be `1.11.0-rc.1`. Looks like the version in `package.json` was bumped after `shrinkwrap-prod/shrinkwrap:prod` was run.

**Comment by dpgraham** [comment]
Thanks for spotting that @john7doe! I'll look into it. I need to figure out how to bump up the shrinkwrap version without re-creating the shrinkwrap altogether.
